### DISCUSSION TITLE
Buffer/block sizes

### DISCUSSION THREAD

**Discussion by vincentark** [apohl79/audiogridder#1016] [Q&A] [answered]
I am relatively new to audiogridder and running the beta12. I have a question about the buffer block size setting. Most of the time I only see as options, none 1024 2048 etc, but sometimes also 512. I know my setup can run even with a 128 buffer (I use Vienna ensemble pro too, which works great with 1 128 buffer, and performs exactly the same as audiogridder at none, but prefer audiogridder).
Is there something I am missing here to get lower buffer settings? I run my DAW, cubase with a RME audio card at 128 buffer.
Thanks for the fantastic program and 
Cheers
Vincent

**Comment by apohl79** [comment, marked as answer]
The block size of the buffers in AG depend on what block size the DAW uses to pass audio samples to the plugin. If the DAW is using 1024 for example, this will be the smallest buffer size in AG.

**Reply by vincentark** [reply]
Ah I see, could Asioguard a steinberg tech to lower cpu load and which dynamicly changes buffersize for not record enabled tracks cause the higher block size in audiogridder? How does audiogridder react to buffer size changes by the host?
I will also investigate further, also by disabling asioguard.
Thanks again
Vincent

**Reply by apohl79** [reply]
Yes the asio guard affects the block size in AG. The host might decide to send audio in smaller chunks than announced. AG has a setting called "Use fixed size outbound buffer". If this is enabled, AG would fill up a buffer until the block size is reached before sending it to the server. Turning that setting off will allow smaller chunks of audio to be send to the server as well (which might have a positive impact on latency).

**Reply by vincentark** [reply]
Disabling asio guard for audiogridder (only, not disable asioguard altogether) fixed the issue 100% and right away I remembered having to do that for vienna ensemble too, but that one would crash horribly with asio guard on, so I did not remember it right away. Performance is very good, I can work and play at a daw buffer of 128 and 1 block also 128 now (with asiogurd disabled) in audiogridder. Performance seems to be a lot better with legacy (@ q1.0 and only differnce checked) then ffmpeg video, is that to be expected? But unless there is animation in the plugin, legacy looks great.
BTW I run audiogridder on a 10th generation 6 core intel laptop @4ghz fixed. I use it for some plugins that need avx or run better with avx since my main daw PC, a quadcore @4ghz has no AVX (overclocked Q9650).
Thanks again and again lol
Vincent

**Reply by apohl79** [reply]
> Performance seems to be a lot better with legacy (@ q1.0 and only differnce checked) then ffmpeg video, is that to be expected?

Interesting. Usually ffmpeg gives better performance. But if legacy works for you, happy days!
